### PR TITLE
remove dao / manager from pillow checkpoint classes

### PR DESCRIPTION
**Pull request by czue**
small-ish prefactor to pillow checkpoints (no functional changes).

now that everything is on django the abstraction layer of the dao is not helpful and was making the code a bit harder to parse.

:fish: or :office: @snopoke cc @biyeun 

wait for tests

### CODE REVIEW

File: corehq/ex-submodules/pillowtop/checkpoints/manager.py [hunk 1]
```diff
@@ -14,82 +14,81 @@
...
-    def __init__(self, dao, checkpoint_id):
-        self._manager = PillowCheckpointManager(dao=dao)
+    def __init__(self, checkpoint_id):
         self.checkpoint_id = checkpoint_id
         self._last_checkpoint = None
 
     def get_or_create(self, verify_unchanged=False):
-        result = self._manager.get_or_create_checkpoint(self.checkpoint_id)
+        result = self._get_or_create(verify_unchanged)
+        return DocGetOrCreateResult(DjangoPillowCheckpoint.to_dict(result.document), result.created)
```
Comment (snopoke) [inline corehq/ex-submodules/pillowtop/checkpoints/manager.py]: does this need to be a `dict` anymore?

**Comment by snopoke** [comment]
Build failing:
```
======================================================================
ERROR: test_basic_cloudant_seq (corehq.apps.hqadmin.tests.test_utils.TestPillowCheckpointSeqStore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/apps/hqadmin/tests/test_utils.py", line 26, in test_basic_cloudant_seq
    pillow_seq_store()
  File "/mnt/corehq/apps/hqadmin/utils.py", line 13, in pillow_seq_store
    db_seq = checkpoint.get_current_sequence_id()
  File "corehq/ex-submodules/pillowtop/checkpoints/manager.py", line 62, in get_current_sequence_id
    return get_or_create_checkpoint(self.checkpoint_id).sequence
AttributeError: 'DocGetOrCreateResult' object has no attribute 'sequence'
```
```
======================================================================
ERROR: test_pillow_reset_checkpoint (pillowtop.tests.test_import_pillows.PillowTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "corehq/ex-submodules/pillowtop/tests/test_import_pillows.py", line 103, in test_pillow_reset_checkpoint
    pillow.set_checkpoint(Change('123', seq_id))
  File "corehq/ex-submodules/pillowtop/pillow/interface.py", line 57, in set_checkpoint
    self.checkpoint.update_to(change['seq'])
  File "corehq/ex-submodules/pillowtop/checkpoints/manager.py", line 68, in update_to
    checkpoint = self._get_or_create(verify_unchanged=True).document
  File "corehq/ex-submodules/pillowtop/checkpoints/manager.py", line 50, in _get_or_create
    result = get_or_create_checkpoint(self.checkpoint_id)
  File "corehq/ex-submodules/pillowtop/checkpoints/manager.py", line 20, in get_or_create_checkpoint
    checkpoint = DjangoPillowCheckpoint.objects.get(checkpoint_id=checkpoint_id)
  File "/vendor/lib/python2.7/site-packages/django/db/models/manager.py", line 92, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 351, in get
    num = len(clone)
  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 122, in __len__
    self._fetch_all()
  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 966, in _fetch_all
    self._result_cache = list(self.iterator())
  File "/vendor/lib/python2.7/site-packages/django/db/models/query.py", line 265, in iterator
    for row in compiler.results_iter():
  File "/vendor/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 701, in results_iter
    for rows in self.execute_sql(MULTI):
  File "/vendor/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 787, in execute_sql
    cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
  File "/vendor/lib/python2.7/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/vendor/lib/python2.7/site-packages/django/db/backends/utils.py", line 65, in execute
    return self.cursor.execute(sql, params)
ProgrammingError: relation "pillowtop_djangopillowcheckpoint" does not exist
LINE 1: ...owtop_djangopillowcheckpoint"."old_sequence" FROM "pillowtop...
```

**Comment by snopoke** [comment]
This simplifies things greatly!

File: corehq/ex-submodules/pillowtop/checkpoints/manager.py (hunk 1 above)
Comment (czue) [inline corehq/ex-submodules/pillowtop/checkpoints/manager.py]: no - was just easier to keep backwards compatible. i got rid of the `dict` version in https://github.com/dimagi/commcare-hq/pull/10408

**Comment by czue** [comment]
:white_check_mark:

**Pull request merged**